packages/flame/lib/src/components/fps_component.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:collection';
+
+import '../../components.dart';
+
+/// The [FPSComponent] is a non-visual component which you can get the current
+/// fps of the game with by calling [fps], once the component has been added to
+/// the component tree.
+class FPSComponent extends Component {
+  FPSComponent({
+    this.windowSize = 60,
+  });
+
+  /// The number of game ticks over which the fps measure will be averaged.
+  final int windowSize;
+
+  /// The queue of the recent game tick durations.
+  /// The length of this queue will not exceed [windowSize].
+  final Queue<double> window = Queue();
```

Comment:
This currently is writeable from the outside, while I love to expose the values, can we somehow make sure people cant add their own values to it?

Reply:
I did that intentionally actually, so that people can calculate other things on the window. Since this should be as efficient as possible I don't want to wrap it in an `UnmodifiableList`, but I should write a comment about that it shouldn't be modfied manually. I think the pinned message in telegram applies here. 😄